crates/pop-drink/src/error.rs

```diff
@@ -0,0 +1,258 @@
+//! A set of errors used for testing smart contracts.
+
+use std::fmt::Debug;
+
+pub use drink::{
+	pallet_assets::Error as AssetsError, pallet_balances::Error as BalancesError,
+	pallet_contracts::Error as ContractsError,
+};
+pub use pop_api::primitives::v0::*;
+use scale::{Decode, Encode};
+
+use crate::utils::decode;
+
+/// Configuration for the runtime error.
+pub trait RuntimeErrorConfig: Debug {
+	/// Error type of the runtime modules.
+	/// Reference: https://paritytech.github.io/polkadot-sdk/master/solochain_template_runtime/enum.RuntimeError.html.
+	type ModuleError: Decode + Encode + Debug;
+	/// Error type of the Pop API depends on the version.
+	type PopApiError: Decode + Encode + Debug + From<u32> + Into<u32>;
+	/// Index of the variant `RuntimeError::Module`. This is based on the index of
+	/// `PopApiError::Module`.
+	const MODULE_INDEX: u8;
+}
+
+/// Runtime error for testing.
+#[derive(Encode, Decode, Debug)]
+pub enum RuntimeError<T: RuntimeErrorConfig> {
+	/// Module errors of the runtime.
+	Module(T::ModuleError),
+	/// Every `PopApiError`.
+	Raw(T::PopApiError),
+}
+
+impl<T: RuntimeErrorConfig> From<RuntimeError<T>> for u32 {
+	/// Converts a `RuntimeError` into a numerical value of `PopApiError`.
+	///
+	/// This conversion is necessary for comparing `RuntimeError` instances with other types
```

Comment:
The rest of these comments dont need to be doc comments imo, just the first line.

Reply:
Resolved